File: app/ui/segment_chain.py
# core/segment_chain.py
from .segment_module import SegmentModule

class SegmentChain:
    def __init__(self, modules):
        self.modules = modules
        self.chars_per_chain = sum(m.chars_per_module for m in modules)

    def clear(self):
        for module in self.modules:
            module.clear()

    def set_text(self, text):
        """
        Teilt den Text auf die Module auf und setzt ihn
        """
        start = 0
        for module in self.modules:
            end = start + module.chars_per_module
            chunk = text[start:end]
            module.set_text(chunk)
            start = end

File: esp/lib/display/SegmentChain.cpp
#include "SegmentChain.h"

SegmentChain::SegmentChain(HT16K33* modules, uint8_t count)
{
    _modules = modules;
    _count = count;
}

void SegmentChain::clear()
{
    for(uint8_t i = 0; i < _count; i++)
        _modules[i].clear();
}


void SegmentChain::setText(const String& text)
{
    for(uint8_t i = 0; i < _count; i++)
    {
        String chunk = text.substring(i * 8, i * 8 + 8);

        while(chunk.length() < 8)
            chunk += " ";

        _modules[i].setText(chunk);
    }
}